packages/amplication-git-utils/src/utils/prepare-files-for-pull-request.ts

```diff
@@ -0,0 +1,82 @@
+import { join } from "path";
+import { Changes } from "octokit-plugin-create-pull-request/dist-types/types";
+import { AMPLICATION_IGNORED_FOLDER } from "../git/git.constants";
+import { AmplicationIgnoreManger } from "./amplication-ignore-manger";
+
+export async function prepareFilesForPullRequest(
+  owner,
+  repositoryName,
+  gitResourceMeta,
+  pullRequestModule
+): Promise<Required<Changes["files"]>> {
+  const amplicationIgnoreManger = new AmplicationIgnoreManger();
+  await amplicationIgnoreManger.init(async (fileName) => {
+    try {
+      const file = await this.getFile({
+        owner,
+        repositoryUrl: repositoryName,
+        path: fileName,
+        baseBranch: undefined, // take the default branch
+      });
+      const { content, htmlUrl, name } = file;
+      console.log(`Got ${name} file ${htmlUrl}`);
+      return content;
+    } catch (error) {
+      console.log("Repository does not have a .amplicationignore file");
+      return "";
+    }
+  });
+
+  //do not override files in 'server/src/[entity]/[entity].[controller/resolver/service/module].ts'
+  //do not override server/scripts/customSeed.ts
+  const doNotOverride = [
+    new RegExp(
+      `^${gitResourceMeta.serverPath || "server"}/src/[^/]+/.+.controller.ts$`
+    ),
+    new RegExp(
+      `^${gitResourceMeta.serverPath || "server"}/src/[^/]+/.+.resolver.ts$`
+    ),
+    new RegExp(
+      `^${gitResourceMeta.serverPath || "server"}/src/[^/]+/.+.service.ts$`
+    ),
+    new RegExp(
+      `^${gitResourceMeta.serverPath || "server"}/src/[^/]+/.+.module.ts$`
+    ),
+    new RegExp(
+      `^${gitResourceMeta.serverPath || "server"}/scripts/customSeed.ts$`
+    ),
+  ];
+
+  const authFolder = "server/src/auth/";
+
+  const files: Required<Changes["files"]> = Object.fromEntries(
```

Comment:
I'm not sure I understand the use of map inside fromEntries also I would break it into composition instead of if's. as I see it you change once the module.path and once the module code, so I would determine the path file and then react inside functions and not if

Reply:
this is an existing logic that I extracted to util function so it's not really a part of this PR